compiler/src/dotty/tools/dotc/Compiler.scala

```diff
@@ -34,6 +34,7 @@ class Compiler {
     List(new Parser) ::             // Compiler frontend: scanner, parser
     List(new TyperPhase) ::         // Compiler frontend: namer, typer
     List(CheckUnused.PostTyper(), CheckShadowing()) :: // Check for unused, shadowed elements
+    List(new WInferUnion) ::        // Winfer-union
```

Comment:
Since you use the mini-phase idea, it must be part of a megaphase, and I think putting it with CheckUnused and CheckShadowing would be fine. It could go in front. That means the megaphase walks the tree once to dispatch to all three miniphases.